doc/http-backoff.md

```diff
@@ -0,0 +1,7 @@
+# HTTP retry and backoff logic
+
+When Ignition is fetching a resource over http(s), if the resource is unavailable Ignition will continually retry to fetch the resource with an exponential backoff between requests.
```

Comment:
I'd write this with a more declarative style and be a bit more concise. Just an opinion.

Ignition can fetch remote `file` contents or Ignition configs (`append` / `replace`) over http and https. 
Ignition will continually retry to fetch resources in the event of timeouts or HTTP 5XX errors. Ignition will wait 10 seconds to receive response headers for a request. Retries use exponential backoff starting by waiting 100 milliseconds  between failed requests and doubling to wait 5 for 5 seconds before retrying.